libraries/AP_GPS/AP_GPS.cpp

```diff
@@ -1034,24 +1034,32 @@ void AP_GPS::Write_DataFlash_Log_Startup_messages()
 
 /*
   return the expected lag (in seconds) in the position and velocity readings from the gps
+  return true if the GPS hardware configuration is known or the delay parameter has been set
  */
-float AP_GPS::get_lag(uint8_t instance) const
+bool AP_GPS::get_lag(uint8_t instance, float &lag_sec) const
 {
     // return lag of blended GPS
-    if (instance == GPS_MAX_RECEIVERS) {
-        return _blended_lag_sec;
+    if (instance == GPS_BLENDED_INSTANCE) {
+        lag_sec = _blended_lag_sec;
+        // auto switching uses all GPS receivers, so all must be configured
+        return all_configured();
     }
 
     if (_delay_ms[instance] > 0) {
         // if the user has specified a non zero time delay, always return that value
-        return 0.001f * (float)_delay_ms[instance];
+        lag_sec = 0.001f * (float)_delay_ms[instance];
+        // the user is always right !!
+        return true;
     } else if (drivers[instance] == nullptr || state[instance].status == NO_GPS) {
-        // no GPS was detected in this instance
-        // so return a default delay of 1 measurement interval
-        return 0.001f * (float)get_rate_ms(instance);
+        // no GPS was detected in this instance so return a default delay of 1 measurement interval
+        lag_sec = 0.001f * (float)get_rate_ms(instance);
+        // check lack of GPS is consistent with user expectation
+        return state[instance].status == NO_GPS;
```

Comment:
Paul and I discussed this directly. I'm going to add an interface to get the worst case lag which we can report here instead. (And provide a mechanism for u-blox to report if it is has confirmed that the lag value it's returning is correct.). Returning the worst case lag term from any possible GPS driver here will leave us with some oversized buffers, but will cover the case of detecting a GPS much later after booting better.